### ISSUE TITLE
Error while reading the pages !  TypeError: "canvas is null"

### ISSUE THREAD

**Issue by danielmaartens** [ansu5555/pdf-viewer-reactjs#35]
HI @ansu5555 @ansuman5555 

This error is happening inconsistently:

Error while reading the pages !
 TypeError: "canvas is null"

I'm loading from a URL. If I copy paste that URL in the browser then it opens the pdf properly. 

What could the possible reasons be for this ?

**Comment by danielmaartens** [comment]
@ansu5555 @ansuman5555 

The question is really why would the canvas be null ?

**Comment by danielmaartens** [comment]
@ansu5555 @ansuman5555 

Funnily enough when I remove the error handling here (see below), the PDF renders properly.
![image](https://user-images.githubusercontent.com/49619761/85291290-9a9a7b00-b49a-11ea-8867-97a9c50fbc57.png)

With error handling:
![image](https://user-images.githubusercontent.com/49619761/85291714-1e546780-b49b-11ea-900d-9295e8b1e647.png)

Error message:
```
Error while reading the pages !
TypeError: "canvas is null"
```

Without error handling:
![image](https://user-images.githubusercontent.com/49619761/85291986-8acf6680-b49b-11ea-9b5e-98ff4621755a.png)

Error message:
```
 Error occured while rendering !
 Error: "Cannot use the same canvas during multiple render() operations. Use different canvas or ensure previous operations were cancelled or completed."
```

**Comment by jaiminmojidra** [comment]
Any update on it? I am also facing same issue.
pdf-viewer-reactjs.js:1 Error occured while rendering !
 Error: Cannot use the same canvas during multiple render() operations. Use different canvas or ensure previous operations were cancelled or completed.
    at InternalRenderTask.initializeGraphics (api.js:2775)
    at api.js:1136

**Comment by amirsaeed671** [comment]
wrapping the pdf component in PureComponent fixed my problem.....

**Comment by ansu5555** [comment]
Fixed in version 2.1.1, it was happening due to multiple times pdf fetch.

If you face the same issue again feel free to open a new issue with error details, steps to reproduce & the pdf file for which its happening.
